src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.inline.hpp

```diff
@@ -121,10 +121,9 @@ inline oop ShenandoahBarrierSet::load_reference_barrier(DecoratorSet decorators,
     return nullptr;
   }
 
-  // Prevent resurrection of unreachable objects that are visited during
-  // concurrent class-unloading.
+  // Allow resurrection of unreachable objects that are visited during concurrent class-unloading.
```

Comment:
Let's not call it "Allow resurrection", which somewhat implies the object has full privileges to exist, i.e. can be inserted into the object graph back. But it really can't. We only do this because we were asked with `AS_NO_KEEPALIVE`.

Something like: "Allow runtime to see unreachable objects that are visited during concurrent class unloading"